### PR TITLE
Data in models

### PR DESCRIPTION
**Pull request by KayLemba**
In this milestone I did the following:
 - Create models
 - Use models to insert data
 - Create custom methods

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'users_id'
+  has_many :likes
+  has_many :comments
+
+  def update_posts_counter
+    post.increment!(:Posts_Counter)
+  end
```
Comment (Emmyn5600) [inline app/models/post.rb]: - [x] I would suggest using one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html)  `after_save` to call the `#update_post_counter` method every time a new post has been saved to the database, then make the `update_post_counter` method private.

Implementation should look like this 

```ruby
 after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

As you can see we are incrementing the post depending on author's post.

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @KayLemba 

Good job so far! I know you have worked hard on making this milestone however, There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight :champagne:

- All linters are passing :heavy_check_mark:
- Good PR description :heavy_check_mark:


## Required Changes :recycle:

- [ ]  Check the comment under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!:clap::clap::clap:

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (KayLemba) [reply app/models/post.rb]: Hello so If I may ask Whats wrong with my Implementation ??

Comment (KayLemba) [reply app/models/post.rb]: Please give me a detailed explaination to why I should make your Implementation. The fact that everything is working fine shoulded be a problem

Comment (KayLemba) [reply app/models/post.rb]: This is not part of the requirements by the way

Comment (KayLemba) [reply app/models/post.rb]: And From the way I look at it. This should be an optional change Im still waiting for a detailed explaination

Comment (KayLemba) [reply app/models/post.rb]: Thanks for review @Emmyn5600 learnt something new from you. DONE!!!!!

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi Kaylo, 
## Congratulations 🎉 Thank you for implementing all the requirements 🙏
### Please go ahead and merge the branch!

## Happy Coding 👏👏👏
#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**